staking.md

```diff
@@ -0,0 +1,663 @@
+- **Staking Cycle Length (`C`)**: 5 epochs (epochs 10 to 14)
+  - **Epoch 10**: Staking Epoch
+  - **Epoch 11**: Entropy Epoch
+  - **Epoch 12**: Leader Election Epoch
+  - **Epoch 13**: Block Production Epoch
+  - **Epoch 14**: Payout Epoch
+- **Participant**: Alice
+- **`MINIMUM_STAKE`**: 100 tokens
+
+### **Alice's Balances**
+
+- **Total Balance (TB)**: 1000 tokens (Alice has already deposited these via `deposit()`).
+- **Available Balance (AB)**: 1000 tokens (initially, no tokens are locked).
+- **Locked Balance (LB)**: 0 tokens.
+
+---
+
+### **Epoch 10 (Staking Epoch)**
+
+#### **Action 1**: Alice Stakes 400 Tokens
+
+- Alice decides to stake **400 tokens** for this staking cycle (epochs **10** to **14**) with production in epoch **13**.
+- She calls `adjustStake(400)`.
+- **Available Balance (AB)** decreases by 400 tokens:
+  - **AB** = 1000 - 400 = **600 tokens**.
+- **Stake Commitment**:
+  - Alice's stake commitment for this staking cycle is now **400 tokens**.
+- **Locked Balance (LB)** remains at **0 tokens** till the end of the epoch (the stake will be locked in Epoch 11).
+
+#### **Action 2**: Alice Withdraws 100 Tokens
+
+- She calls `withdraw(100)`.
+- **Total Balance (TB)** decreases by 100 tokens:
+  - **TB** = 1000 - 100 = **900 tokens**.
+- **Available Balance (AB)** decreases by 100 tokens:
+  - **AB** = 600 - 100 = **500 tokens**.
+- **Locked Balance (LB)** **400 tokens**.
+
+---
+
+### **Epoch 11 (Entropy Epoch)**
+
+- Alice can now plan her stake for the **following** staking cycle and decides on no change.
+
+- **System Process**:
+  - The system awaits the parent chain's hash to generate entropy.
+- **Alice's Balances**:
+  - No changes occur in this epoch.
+  - **AB** remains at **500 tokens**.
+  - **LB** remains at **400 tokens**.
+
+---
+
+### **Epoch 12 (Leader Election Epoch)**
+
+- Alice's stake of **400 tokens** stays **locked** for the duration of the staking cycle (epochs **12** to **14**).
+- **Alice's Balances**:
+  - **LB** = **400 tokens**.
+  - **AB** = **500 tokens**.
+
+#### **Adjusting Stake for Next Cycle**
+- Alice can now plan her stake for the next next staking cycle (block production in epoch 15) and decides on no change.
+- She decides to **withdraw** her stake after the current cycle:
+  - She calls `adjustStake(-400)`.
+  - This action schedules her stake to be reduced by 400 tokens after the current cycle ends.
+- **Note**: The **AB** and **LB** remain unchanged until the stake is unlocked.
+
+---
+
+### **Epoch 13 (Block Production Epoch)**
+
+- **Block Production**:
+  - Alice participates in block production based on her **locked stake** of **400 tokens**.
+- **Alice's Balances**:
+  - No changes occur in this epoch.
+  - **AB** remains at **500 tokens**.
+  - **LB** remains at **400 tokens**.
+
+---
+
+### **Epoch 14 (Payout Epoch)**
+
+#### **Rewards Distribution**
+
+- Alice receives rewards based on her participation, paid out directly to her account.
```

Comment:
I think rewards will be payed out to a/the contract or else the delegated staking will not work?

Reply:
Yes, if you are a "delegated" staker your delegation contract should be the one calling the deposit and stake functions and payments of rewards should go there.

Reply:
I added some info on Alice.